napari/layers/points/points.py

```diff
@@ -1098,16 +1101,27 @@ def mode(self, mode):
             return
         old_mode = self._mode
 
+        if old_mode == Mode.ADD:
+            self.mouse_drag_callbacks.remove(add)
+        elif old_mode == Mode.SELECT:
+            # add mouse drag and move callbacks
+            self.mouse_drag_callbacks.remove(select)
+            self.mouse_move_callbacks.remove(highlight)
```

Comment:
Same comments as with the labels layer: it would be good to use a dictionary, or some kind of stack, so that you could remove the keybindings from the previous mode change, without an if/else cascade that goes through all the modes — which I expect will be super painful for Shapes!

Reply:
As discussed in the labels review, I'd rather this wait for the refactor in #1048